src/language_concepts/data_types.md

```diff
@@ -1,206 +1,192 @@
+All values in Noir are fundamentally `Field` elements. For a more approachable developing experience, abstractions are added on top to introduce different data types in Noir.
 
-A **private value** is known only to the Prover, while, a **public value** is known by the Prover and Verifier.
-All public and private values are primitive types, though individual
-fields of a struct may also be marked private or public.
+Noir has two category of data types: primitive types (e.g. `Field`, integer, `bool`) and compound types (e.g. array, tuple, struct). Each value can either be private or public.
 
-Note that the terms private and public when applied to a type (e.g. in `pub Field`) have a different meaning than when
-applied to a function `pub fn foo() { ... }`.
-The later is a visibility modifier familiar to those coming from existing
-langauges, while the former can be thought of as a visibility modifier for the Verifier.
-So a `pub Field` would be visible to the Verifier while a `Field` would not (private is the default modifier).
+## Private & Public Types
 
-## Primitive Types
-
-Any of these types can be private (the default) or public (via a `pub` modifier).
-
-Note that private types are also referred to as witnesses.
+A **private value** is known only to the Prover, while a **public value** is known by both the Prover and Verifier. All public and private values are primitive types, though individual fields of compound types may also be marked private or public.
 
-#### Field Type
+> **Note:** For Noir programs using public smart contract verifiers, public values defined in such programs can be considered as public knowledge once their proofs are verified on-chain.
 
-A field type corresponds to a native field type in the backend. Usually this is a (roughly) 256-bit integer.
-This should generally be the default type reached for to solve problems. Using a smaller integer type like `u64` incurs
-extra range constraints and so is less efficient rather than more.
-
-```rust,noplaypen
-fn main(x : Field, y : Field) {
-    let z = x + y;
-}
-```
+Public data types are treated no differently to private types apart from the fact that their values will be revealed in proofs generated. Simply changing the value of a public type will not change the circuit (where the same goes for changing values of private types as well).
 
-`x`, `y` and `z` are witness types (private `Field`). Using the `let` keyword we derived a new witness `z` which is constrained to be equal to `x + y`.
+_Private_ are also referred to as _witnesses_ sometimes.
 
-#### Integer Type
+> **Note:** The terms private and public when applied to a type (e.g. `pub Field`) have a different meaning than when applied to a function (e.g. `pub fn foo() {}`).
+>
+> The latter is a visibility modifier familiar to those coming from other languages, while the former can be thought of as a visibility modifier for the Verifier to interpret.
```

Comment:
Can we re-order this so that the former is explained first? I think it'd be less confusing. If we want to introduce the more familiar function visibility first we should re-order the first sentence on line 19 then.